Source/items.cpp

```diff
@@ -3428,51 +3430,51 @@ void PrintItemPower(char plidx, ItemStruct *x)
 	switch (plidx) {
 	case IPL_TOHIT:
 	case IPL_TOHIT_CURSE:
-		sprintf(tempstr, _("chance to hit: %+i%%"), x->_iPLToHit);
+		strcpy(tempstr, fmt::format(_("chance to hit: {:+}%"), x->_iPLToHit).c_str());
 		break;
 	case IPL_DAMP:
 	case IPL_DAMP_CURSE:
-		sprintf(tempstr, _("%+i%% damage"), x->_iPLDam);
+		strcpy(tempstr, fmt::format(_("{:+}% damage"), x->_iPLDam).c_str());
 		break;
 	case IPL_TOHIT_DAMP:
 	case IPL_TOHIT_DAMP_CURSE:
-		sprintf(tempstr, _("to hit: %+i%%, %+i%% damage"), x->_iPLToHit, x->_iPLDam);
+		strcpy(tempstr, fmt::format(_("to hit: {:+}%, {:+}% damage"), x->_iPLToHit, x->_iPLDam).c_str());
 		break;
 	case IPL_ACP:
 	case IPL_ACP_CURSE:
-		sprintf(tempstr, _("%+i%% armor"), x->_iPLAC);
+		strcpy(tempstr, fmt::format(_("{:+}% armor"), x->_iPLAC).c_str());
 		break;
 	case IPL_SETAC:
-		sprintf(tempstr, _("armor class: %i"), x->_iAC);
+		strcpy(tempstr, fmt::format(_("armor class: {}"), x->_iAC).c_str());

```

Comment:
One consideration I hadn't made is that this actually removes some context for the trasnlators as they now cannot see if a string or a number is being inserted. We might want to explicitly use `{:d}` when ever the value is numeric.

Reply:
Ah, that is unfortunate. For this PR I have applied a series of regex replacements followed by manual fixes, that means that the initial information (`%i`, `%d`, etc.) has been lost. To replace `%d` with `{:d}` for instance I have to either redo the whole process, or look at the diff and replace `{}` with `{:d}` at the right places.

I think for translations the best thing to do would be to use named insertions, like `{gold}` instead of using `{}` or `{:d}`, but that's a manual operation and a lot of tedious work.

Reply:
Yeah I think naming should be done more selective as it appears to require a few other manual change an probably isn't necessary in most cases. Would you be willing to apply the regex to change `%i `to `{:d}`? Else I would do it manually when merging.

Reply:
> Yeah I think naming should be done more selective as it appears to require a few other manual change an probably isn't necessary in most cases.

Makes sense.

> Would you be willing to apply the regex to change %i to {:d}? Else I would do it manually when merging.

I have squashed multiple commits (one per major step) into one before pushing, but I can maybe come back to the state prior to that rebase using the git reflog. That would prevent me from having to restart whole PR from scratch. I'll try that.

Reply:
Thanks, reflog can be a life saver :)

Reply:
Should be better now. I haven't rebased/squashed the commits for now in case something needs updating between one of the steps, but can do once the PR is accepted.